testing/materialized_views.test

```diff
@@ -367,4 +367,21 @@ do_execsql_test_on_specific_db {:memory:} matview-mixed-operations-sequence {
 200|100|1
 100|25|1
 200|100|1
-300|150|1}
\ No newline at end of file
+300|150|1}
+
+do_execsql_test_on_specific_db {:memory:} matview-projections {
+    CREATE TABLE t(a,b);
+
+    CREATE MATERIALIZED VIEW v AS
+        SELECT b, a, b + a as c , (b * a) + 10 as d , min(a,b) as e
+        FROM t
+        where b > 2;
+
+    INSERT INTO t VALUES (1, 1);
+    INSERT INTO t VALUES (2, 2);
+    INSERT INTO t VALUES (3, 4);
+    INSERT INTO t VALUES (4, 3);
+
+    SELECT * from v;
```

Comment:
Regarding the commit message about the previous tests having been wrong: there's a good opportunity here, while this is still quite early, to start differential fuzzing materialized views, and it's quite easy too:

```rust
assert_eq!(select * from my_matview, select * from (<my_matview_definition>)
```

We are missing a good "arbitrary sql generator" at the moment, but the plumbing can probably be lifted from some of the existing fuzz tests

Reply:
What is in scope for "the impl"? Our initial approach to testing in tursodb was that we supported the query translation / btree / vdbe modules basically with only handcrafted TCL tests and Rust unit tests, and once we finally started adding fuzz tests we discovered that, in every case, etither 1. there were a ton of edge cases not covered or 2. the feature did not work at ALL. Even today the fuzz coverage is bad because it's been retroactively bolted on instead of built with the feature set. So the earlier we do it the better.

Reply:
Personally, until we implement circuits and joins, I think we should be ready to throw away all of this code completely. The value of fuzz testing is then diminished.

Obviously I am hoping we are on the right track and will keep all of it.